### PR TITLE
Handle errors of db.Begin() in DB.inTransaction()

### PR DESCRIPTION
**Pull request by zenovich**

### CODE REVIEW

File: app/database/db.go [hunk 1]
```diff
@@ -37,24 +37,26 @@ func Open(source interface{}) (*DB, error) {
 }
 
 func (conn *DB) inTransaction(txFunc func(*DB) error) (err error) {
+	var transactionStarted = false
 	var txDB = conn.db.Begin()
-	if err != nil {
-		return err
+	if txDB.Error != nil {
```
Comment (smadbe) [inline app/database/db.go]: Haha it was so stupid 👍  (due to a change to the initial code). Such a shame that Go does not help spotting this kind of mistake.

File: app/database/db.go [hunk 2]
```diff
@@ -37,24 +37,26 @@ func Open(source interface{}) (*DB, error) {
...
+		return txDB.Error
 	}
+	transactionStarted = true
 	defer func() {
 		if p := recover(); p != nil {
 			// ensure rollback is executed even in case of panic
-			txDB.Rollback()
+			if transactionStarted {
```
Comment (smadbe) [inline app/database/db.go]: How can we arrive here with `! transactionStarted` as the `defer` is defined after the "begin/return error"?

Comment (zenovich) [reply app/database/db.go]: This function is a closure, so it sees all the variables from the scope that defines it.

Comment (zenovich) [reply app/database/db.go]: You're right, we cannot. I'll fix it right now

**Pull request merged**